src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -441,7 +441,9 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
             Models.RazorSemanticTokensEdit[] edits)
         {
             var updatedTokens = previousCSharpTokens.ToList();
-            foreach (var edit in edits)
+
+            // C# returns edits that are expected to be applied in reverse order.
+            foreach (var edit in edits.Reverse())
```

Comment:
Also, `.Reverse()` is a performance concern, if we kept this I would say it should be `for(int i = edits.Length; i >=0;  i--)`.